discovery_engine_core/tooling/src/bin/backend/routes.rs

```diff
@@ -171,13 +171,13 @@ async fn fetch_popular_results(
     });
 
     // after the first search also include search_after
-    let index = app_state.index.read().await;
     let from_index = app_state.from_index.read().await;
-    if *index > 0 || !from_index.is_empty() {
```

Comment:
:question: Shouldn't we still check for `index > 0`?

Reply:
I've added that back just in case, but `from_index` should be set to `None` for the "first page" call if I understand correctly